spec/datadog/profiling/spec_helper.rb

```diff
@@ -72,8 +72,14 @@ def samples_for_thread(samples, thread)
     samples.select { |sample| object_id_from(sample.labels.fetch(:'thread id')) == thread.object_id }
   end
 
-  def build_stack_recorder
-    Datadog::Profiling::StackRecorder.new(cpu_time_enabled: true, alloc_samples_enabled: true)
+  # We disable heap_sample collection by default in tests since it requires some extra mocking/
+  # setup for it to properly work.
+  def build_stack_recorder(heap_samples_enabled = false)
```

Comment:
Minor: It's nice to use keyword arguments for methods with boolean to make them self-documenting (e.g. `build_stack_recorder(heap_samples_enabled: false)` vs `build_stack_recorder(false)`).

Of course maybe your setup shows the arg names anyway, ehehe I should get it working as well (and it would be amazing if github did more of that......).

```suggestion
  def build_stack_recorder(heap_samples_enabled: false)
```